### PR TITLE
Datasets handler storage

### PR DESCRIPTION
**Pull request by Moasib-Arif**

### CODE REVIEW

**Review by eldeal — APPROVED** [review]
Looks good to me, but probably worth just giving Jo a heads up before we merge this into his working branch. One to discuss at standup

File: Pipfile
Comment (mikeAdamss) [inline Pipfile:11]: we don't need this here. We probabaly need to update the readme but the way it works is:

- `pipenv install` - installs the libraries in packages
- `pipenv install -d` (d for dev) installs the libraries in "packages" **and** the libraries in "dev-packages".

You can just use the latter and remove this duplicate pytest entry from "packages".

It's a separation principle, we want "packages" to only contain libraries used when running in production, keeps things nice and clean.

Could you likewise move jsonschema down to dev-packages as well (and confirm the tests run ok please).

File: Pipfile
Comment (mikeAdamss) [inline Pipfile:10]: Can you keep with SPARQLWrapper please. I know it's odd but we had differing behaviour of some kind between "sparqlwrapper" and "SPARQLWrapper" on the other branch.

Could have been something else going on (they seem the same on the surface) but if we change to the lower case install it'd be better to do it after it all gets merged together (better safe than sorry).

File: src/main.py
Comment (mikeAdamss) [inline src/main.py:6]: I like the use of a constant (good thinking) but:

- python contants are uppercase, `JSONLD` please.
- given this constant doesn't naturally sit in either file that's using it can you make a `constants.py` at the same level as `main.py` and import it from there please. Nice and obvious what it is then and encourages future folks to follow the same pattern with other mimetypes.

File: tests/test_datasets.py
Comment (mikeAdamss) [inline tests/test_datasets.py:29]: use your constant  😄

File: src/main.py
Comment (mikeAdamss) [inline src/main.py:17]: there's a flask "jsonify" function for returning json objects, probably worth a quick investigation: https://www.geeksforgeeks.org/use-jsonify-instead-of-json-dumps-in-flask/

**Review by mikeAdamss — COMMENTED** [review]
this is looking pretty grand - though I added a few extra comments while catching up, let me know if you have any questions 👍

**Comment by mikeAdamss** [comment]
putting this on hold as we've had a change in direction. May or may not close in time but keep the example code around for a little bit while we figure that out.

do not merge or work on.

**Comment by mikeAdamss** [comment]
closing as we've done a pivot.

**Pull request closed without merging**
